### DISCUSSION TITLE
GitHub Codespaces in VSCode Desktop

### DISCUSSION THREAD

**Discussion by maxfalguera** [community/community#51929] [New to GitHub] [answered]
Hi, i'm having troubles signing in to VsCode Desktop with the synced settings of my github codespaces that i've been using online. I think it fails with the GitHub Autentication. I need to sign in in the desktop version because there's an extension that is not able for online version. Thanks!

**Comment by SMelidoni** [comment, marked as answer]
Hi there, 

I'm sorry to hear that you're having trouble signing in to VSCode Desktop with the synced settings of your GitHub Codespaces. Let's try to troubleshoot the issue together.

First, make sure that you have the latest version of VSCode Desktop installed. You can download it from the official website.

Next, ensure that you have signed in to VSCode Desktop with the same GitHub account that you use for your GitHub Codespaces. You can sign in by clicking on the "Sign in" button in the lower-left corner of the VSCode window.

If you have already signed in to VSCode Desktop with the correct GitHub account, try resetting the sync feature by following these steps:

1. Open VSCode Desktop and click on the gear icon in the lower-left corner of the window.
2. Click on "Settings" and then "Sync" in the left-hand menu.
3. Click on the "Reset Sync Data" button at the bottom of the page.
4. Once the sync data has been reset, sign in again to your GitHub account using the "Sign in with GitHub" button in the VSCode window.

If resetting the sync feature doesn't work, you can try clearing your VSCode Desktop cache by following these steps:

Windows:
1. Close VSCode Desktop
2. Open the Run dialog box by pressing the Windows key + R
3. Type "%APPDATA%\Code" and press Enter
4. Delete the contents of the "CachedData" and "Cache" folders
5. Restart VSCode Desktop and try signing in again

Mac:
1. Close VSCode Desktop
2. Open Finder
3. Click on "Go" in the menu bar at the top of the screen
4. Click on "Go to Folder" from the dropdown menu, or press the "Shift + Command + G" keyboard shortcut
5. In the "Go to Folder" dialog box, type "~/Library/Application Support/Code" and click on the "Go" button
6. Delete the contents of the "CachedData" and "Cache" folders
7. Restart VSCode Desktop and try signing in again

If none of these solutions work, please provide more details about the issue you're experiencing, including any error messages you're receiving, and we can try to troubleshoot further.

**Reply by maxfalguera** [reply]
Hey Simone, thanks to your advice i managed to fix it. Thanks a lot!!

**Reply by SMelidoni** [reply]
Great to hear! Would you mind marking it as the accepted solution so that others can easily find the solution to this problem? To do so, you can click on the 'Mark as solution' button underneath my answer. Thank you!

**Comment by velsietis** [comment]
I had issues connecting to a Codespace with VS Code, but thought I'd add a note to say what solved it for me:

From the Command Palette (Ctrl+P), make sure to choose `Codespaces: Sign in` before trying `Codespaces: Connect to Codespace...`.

It seems that although the "Connect to Codespace" command does redirect me to log-in, there's some kind of bug that makes it get stuck in a loop and I end up with a "Having trouble signing in..." notification.
